compiler/testData/codegen/box/primitiveTypes/kt2269.kt

```diff
@@ -1,13 +1,16 @@
 // IGNORE_BACKEND: JVM_IR
 fun box() : String {
-    230?.toByte()?.hashCode()
-    9.hashCode()
+    // Just hard enough that the test won't get optimized away at compile time.
```

Comment:
Were there any issues with this tests involving constant folding?

Reply:
With this pass and some branch optimizations by Mads, the test simply returns OK :) The conditions are optimized away:
```
...
LDC "OK"
ARETURN
```

Reply:
@madsager seems it's a bug in your optimizer:
```
fun box() {
    val y: String?
    if (1 == 0) { //no nop and linenumber in bytecode        
        y = "123"
    }
    else {
        y = null
    }    
}
```

Reply:
Thanks Mike, I agree. Ting-Yuan, we should work together on finding a way to fix that.

Reply:
I've disabled these 2 tests and let's address it in another PR. This one is probably too large already.